CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+
+- Our [lack of line numbers is a feature](https://highlightjs.readthedocs.io/en/latest/line-numbers.html). The simple case isn't hard to add though with a little custom JavaScript.
+- We [embrace plug-ins and extensions](https://github.com/highlightjs/highlight.js/issues/2225) rather than adding lots of tiny features and configuration options.
+- We are much more than a keyword highlighter. We do make attempts to understand context...
+- ...Yet we are not a full parser for any language, we're still just a fancy nested pattern matcher.
+- Our [auto-detect is sadly not powered by magic unicorn dust](https://github.com/highlightjs/highlight.js/issues/1213). It's best effort, not best in class.
+
+
+## Reporting an Issue
+(Misc.)
+- If you found a language highlighting issue, use [this JSFiddle](https://jsfiddle.net/ajoshguy/2bmdswn6/) to create a test case.
+
+
+## Fixing an Issue (PRs)
+If you're new to contributing to open-source, have a look at [this GitHub Guide](https://guides.github.com/activities/forking). It explains the general process of GitHub collaboration.
+
+If you feel comfortable with our [prerequisites](#prerequisites), you can grab any issue [marked `beginner friendly`](https://github.com/highlightjs/highlight.js/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22).
+
+
+### Rules
+Don't worry; these rules are pretty standard.
+
+Please open an issue before your PR, so we can explore and discuss the topic at hand. Your time is important, and we need to be sure it's well-spent.
+
+**Before** you start coding, keep these tips in mind:
+- You should usually add markup tests (ie. whenever you've made a significant grammar change or fixed a bug)
+- Change only what needs to be changed; don't re-lint/rewrite whole files when fixing bugs
+- Linting needs a dedicated commit
+
+**When you're ready** to open a PR, we have some housekeeping tasks for you.
+- Find and link to the relevant issue
+- Add a changelog entry to [CHANGES.md](https://github.com/highlightjs/highlight.js/blob/master/CHANGES.md)
+- If necessary, add your name to [AUTHORS.txt](https://github.com/highlightjs/highlight.js/blob/master/AUTHORS.txt) under Contributors
+
+
+## Setup and Run
+
+You'll of course need Node.js and npm or yarn.
+
+Often when contributing a PR (that doesn't make any specific changes to browser
+features) it's more than sufficient to build and test only the Node.js build.
+Our CI process will guarantee that the browser build is still green.
+
+**Should we shown yarn examples also?**
+
+```
+npm run build
+npm run test
+```
+
+The browser library must be built and tested separately:
+
+```
+npm run build-browser
+npm run test-browser
+```
+
+**Where should we mention the developer tool? ./tools/developer.html**
+
+The "run" here made me think of that because otherwise there isn't really
+much to "run" (other than tests).
+
+
+
+For more details see [Building and Testing](https://highlightjs.readthedocs.io/en/latest/building-testing.html).
```

Comment:
Tiny aesthetics.
```suggestion
For more details see the docs on [Building and Testing](https://highlightjs.readthedocs.io/en/latest/building-testing.html).
```

Reply:
Ok let's not run these tiny one offs by me... please fix grammar, tweak typos, rewrite small sections as necessary... take the initiative, whatever.  I'm trying to focus more on the 1 mile high view. :-)